rmw_connext_shared_cpp/src/qos.cpp

```diff
@@ -156,34 +162,81 @@ bool
 get_datareader_qos(
   DDS::DomainParticipant * participant,
   const rmw_qos_profile_t & qos_profile,
+  const char * namespace_,
+  const char * node_name,
+  const char * dds_topic_name,
   DDS::DataReaderQos & datareader_qos)
 {
-  DDS::ReturnCode_t status = participant->get_default_datareader_qos(datareader_qos);
-  if (status != DDS::RETCODE_OK) {
-    RMW_SET_ERROR_MSG("failed to get default datareader qos");
-    return false;
+  bool profile_found = false;
+  if (are_node_profiles_allowed()) {
+    DDS::DomainParticipantFactory * dpf = DDS::DomainParticipantFactory::get_instance();
+    if (!dpf) {
+      RMW_SET_ERROR_MSG("failed to get participant factory");
+      return false;
+    }
+    char fqnn[RMW_NAMESPACE_MAX_LENGTH + RMW_NODE_NAME_MAX_NAME_LENGTH];
```

Comment:
Account for null character?
```suggestion
    char fqnn[RMW_NAMESPACE_MAX_LENGTH + RMW_NODE_NAME_MAX_NAME_LENGTH + 1];
```

Reply:
Good catch